genson/src/main/java/com/owlike/genson/convert/DefaultConverters.java

```diff
@@ -1337,4 +1337,150 @@ public T deserialize(ObjectReader reader, Context ctx) throws Exception {
       return value;
     }
   }
+
+  public static class OptionalConverterFactory implements Factory<Converter<Optional<?>>> {
+    public static final OptionalConverterFactory instance = new OptionalConverterFactory();
+    private OptionalConverterFactory() {
+    }
+
+    @Override
+    public Converter<Optional<?>> create(final Type type, final Genson genson) {
+      Type optionalType = typeOf(0, type);
+      //noinspection unchecked
+      return new OptionalConverter(genson.provideConverter(optionalType));
+    }
+  }
+
+  @HandleClassMetadata
+  @HandleBeanView
+  public static class OptionalConverter<T> implements Converter<Optional<T>> {
+
+    private Converter<T> converter;
+
+    public OptionalConverter(Converter<T> converter) {
+      this.converter = converter;
+    }
+
+    @Override
+    public void serialize(final Optional<T> optional, final ObjectWriter writer, final Context ctx) throws Exception {
+      writer.beginObject();
+      writer.writeMetadata("class", Optional.class.getName());
+      if (optional.isPresent()) {
+        writer.writeName("value");
+        converter.serialize(optional.get(), writer, ctx);
+      }
+      writer.endObject();
+    }
+
+    @Override
+    public Optional<T> deserialize(final ObjectReader reader, final Context ctx) throws Exception {
+      reader.beginObject();
+      while (reader.hasNext()) {
+        reader.next();
+        if ("value".equals(reader.name())) {
+          return Optional.of(converter.deserialize(reader, ctx));
+        }
+      }
+      return Optional.empty();
+    }
+  }
+
+  @HandleClassMetadata
+  public static class OptionalIntConverter implements Converter<OptionalInt> {
+    public final static OptionalIntConverter instance = new OptionalIntConverter();
+
+    private OptionalIntConverter() {
+    }
+
+    @Override
+    public void serialize(final OptionalInt optional, final ObjectWriter writer, final Context ctx) throws Exception {
+      writer.beginObject();
+      writer.writeMetadata("class", OptionalInt.class.getName());
+      if (optional.isPresent()) {
+        writer.writeName("value");
+        Converter<Integer> converter = ctx.genson.provideConverter(Integer.TYPE);
```

Comment:
Why do you need a converter? `writer.writeValue(optional.getAsInt())` should be sufficient and much more efficient, as we know at this point what the value is.